staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -975,6 +975,10 @@ func (c *Cacher) GuaranteedUpdate(
 
 // Count implements storage.Interface.
 func (c *Cacher) Count(pathPrefix string) (int64, error) {
+	cnt := len(c.watchCache.ListKeys())
```

Comment:
if we're going to ask the store for a count, it makes more sense to add a Count() function to store than to make it allocate and return all keys just for us to count them

Reply:
I doesn't fully get  your meaning. 

Do you mean that we should move this to `watchCache`, make it looks like `func (w *watchCache) Count() int { return len(c.watchCache.ListKeys()) }`, or change the underlay `watchCache.store` and add a `Count()` method to `cache.Indexer` interface?

Reply:
I mean add a Count method to store so it can return a count directly without allocating / iterating / returning all keys

Reply:
I was concerned that the implications of changing the underlay `cache.Interface` is too big, but your point is also really reasonable. Therefore, I was very conflicted all day yesterday.

But I just saw an interesting PR https://github.com/kubernetes/kubernetes/pull/126754 that re-reimplements a new store that is totally different with `cache.Indexer`. So I think there is a way to have the best of both side, by relying on this PR and avoiding the need to modify `cache.Indexer`.

Reply:
I think we should avoid to use `ListKeys`, it has a lock and will consume a lot of memory.

Reply:
I agree I don't think we should switch this to call ListKeys